test/contact-messages.js

```diff
@@ -1,101 +1,96 @@
 const tape = require('tape')
+const run = require('promisify-tuple')
 const u = require('./util')
 
 const feedId = '@th3J6gjmDOBt77SRX1EFFWY0aH2Wagn21iUZViZFFxk=.ed25519'
 
-tape('follow', t => {
+tape('friends.follow()', async t => {
   const sbot = u.Server({ tribes: true })
 
   /* FOLLOW */
-  sbot.friends.follow(feedId, {}, (err, msg) => {
-    if (err) throw err
+  const [err1, msg1] = await run(sbot.friends.follow)(feedId, {})
+  t.error(err1)
 
-    // NOTE get here just to confirm recps: undefined not present
-    sbot.get(msg.key, (_, value) => {
-      t.deepEqual(
-        value.content,
-        {
-          type: 'contact',
-          contact: feedId,
-          following: true
-        },
-        'publishes a follow message!'
-      )
+  // NOTE get here just to confirm recps: undefined not present
+  const [err2, value] = await run(sbot.get)(msg1.key)
+  t.error(err2)
+  t.deepEqual(
+    value.content,
+    {
+      type: 'contact',
+      contact: feedId,
+      following: true
+    },
+    'publishes a follow message!'
+  )
 
-      /* UNFOLLOW */
-      sbot.friends.follow(feedId, { state: false }, (err, msg) => {
-        if (err) throw err
+  /* UNFOLLOW */
+  const [err3, msg3] = await run(sbot.friends.follow)(feedId, { state: false })
+  t.error(err3)
+  t.deepEqual(
+    msg3.value.content,
+    {
+      type: 'contact',
+      contact: feedId,
+      following: false,
+      recps: undefined
+    },
+    'publishes a unfollow message!'
+  )
 
-        t.deepEqual(
-          msg.value.content,
-          {
-            type: 'contact',
-            contact: feedId,
-            following: false,
-            recps: undefined
-          },
-          'publishes a unfollow message!'
-        )
+  /* PRIVATE FOLLOW */
+  const [err4, msg4] = await run(sbot.friends.follow)(feedId, { recps: [sbot.id] })
+  t.error(err4)
+  t.match(msg4.value.content, /box\d$/, 'publishes a private follow')
 
-        /* PRIVATE FOLLOW */
-        sbot.friends.follow(feedId, { recps: [sbot.id] }, (err, msg) => {
-          if (err) throw err
-          t.match(msg.value.content, /box\d$/, 'publishes a private follow')
-
-          sbot.close(t.end)
-        })
-      })
-    })
-  })
+  await run(sbot.close)()
```

Comment:
can we do sbot.close(t.end()) here instead?

Reply:
I could, but does it make it better? I was hoping that people could just read code "sequentially" (as if it were synchronous control flow), so each "step" would be on its own line of code.

Reply:
yeah it follows the pattern of the rest of the file now. It's just a bit verbose :) But either way works.